docs/tutorials/plugin-provider.md

```diff
@@ -0,0 +1,15 @@
+# Plugin provider
+
+The Plugin provider is a provider that allows DNS providers to integrate with ExternalDNS via an HTTP interface. The Plugin provider implements the Provider interface but instead of implementing code specific to a provider, it implements and HTTP client that sends request to an HTTP API. The idea behind this is that providers can be implemented in a separate program that exposes an HTTP API that the Plugin provider can interact with. The ideal setup for providers is to run as sidecars of the ExternalDNS container, but this is not strictly a requirement.
+
+## Architectural diagram
+
+![Pluging provider](../img/plugin-provider.png)
+
+## API guarantees
+
+Providers implementing the HTTP API have to keep in sync with changes to the Go types `plan.Changes` and `endpoint.Endpoint`. We do not expect to make significant changes to those types given the maturity of the project, but we can't exclude that changes will need to happen. We commit to publishing changes to those in the release notes, to ensure that providers implementing the API can keep providers up to date quickly.
```

Comment:
Have you considered introducing a versioning of the endpoint the plugin provider (e.g. localhost:8888/v1/dns) is hitting to be able to provide a compatibility layer?

Reply:
As far as I know API's have better ways of handling versioning than a path.
http://opensource.zalando.com/restful-api-guidelines/#compatibility and especially
http://opensource.zalando.com/restful-api-guidelines/#114

Reply:
@mrueg good point, I was never a fan of versions in the URL. What @szuecs is proposing makes sense to me (especially having worked with those API guidelines before) so I am happy to implement that.

Reply:
Thanks for sharing @szuecs! Negotiating by content-type sounds good to me as well.